target/bin/setquota

```diff
@@ -34,23 +34,19 @@ fi
 
 # Protect config file with lock to avoid race conditions
 touch "${DATABASE}"
-(
-  flock -e 200
-
-  if [ -z "${QUOTA}" ]; then
-    read -r -s "Enter quota (e.g. 10M): " QUOTA
-    echo
-    [[ -z "${QUOTA}" ]] && errex "Quota must not be empty. Use 0 for unlimited quota"
-  fi
-
-  # check quota
-  if [[ -n ${QUOTA} ]] && ! echo "${QUOTA}" | grep -qE "^([0-9]+(B|k|M|G|T)|0)\$"
-  then
-    usage
-    errex "invalid quota format. e.g. 302M (B (byte), k (kilobyte), M (megabyte), G (gigabyte) or T (terabyte))"
-  fi
-
-  delquota "${USER}"
-  echo "${USER}:${QUOTA}" >>"${DATABASE}"
-) 200< "${DATABASE}"
+create_lock "$(basename "$0")"
+if [ -z "${QUOTA}" ]; then
```

Comment:
Just mentioning this here so you don't have to search through all the code in the end. Just a minor "issue"